.github/workflows/code-qa.yml

```diff
@@ -59,57 +29,99 @@ jobs:
       - name: Run Pre-Commit
         run: uv run pre-commit run --all-files
 
-  build:
-    name: Build wheel
-    runs-on: ubuntu-latest
+  build: # {{{1
+    name: Build wheel for ${{matrix.os}} / ${{matrix.arch}}${{ matrix.manylinux && ' / ' || '' }}${{ matrix.manylinux || '' }}
+    runs-on: ${{matrix.os}}
+    strategy:
+      fail-fast: false
+      matrix:
+        include:
+          # Linux manylinux {{{
+          - os: ubuntu-latest
+            arch: x86_64
+            manylinux: manylinux
+          - os: ubuntu-24.04-arm
+            arch: aarch64
+            manylinux: manylinux
+          # }}}
+          # Linux musllinux {{{
+          - os: ubuntu-latest
+            arch: x86_64
+            manylinux: musllinux
+          - os: ubuntu-24.04-arm
+            arch: aarch64
+            manylinux: musllinux
+          # }}}
+          # macOS {{{
+          - os: macos-13
+            arch: x86_64
+          - os: macos-14
+            arch: arm64
+          # }}}
+          # Windows {{{
+          - os: windows-latest
+            arch: x86
+          - os: windows-latest
+            arch: AMD64
+          # }}}
+    env:
+      CIBW_ARCHS: ${{matrix.arch}}
+      CIBW_ENVIRONMENT: 'PATH="$HOME/.cargo/bin:$PATH"'
+      CIBW_ENVIRONMENT_WINDOWS: 'PATH="$UserProfile\.cargo\bin;$PATH"'
+      MACOSX_DEPLOYMENT_TARGET: '10.13'
     steps:
       - uses: actions/checkout@v4
         with:
           fetch-depth: 0
-      - uses: astral-sh/setup-uv@v5
-        with:
-          enable-cache: true
-          cache-dependency-glob: "**/pyproject.toml"
-          python-version: "3.12"
-      - name: Build packages
-        run: uv build
-      - name: Verify packages
-        run: uvx twine check dist/*
+      - name: Determine which Python versions to test against
+        shell: bash
+        env:
+          ENV_TAG: ${{runner.os}}-${{matrix.arch}}-${{matrix.manylinux}}
+          MANYLINUX: ${{matrix.manylinux}}
+        run: |-
+          if [[ "$ENV_TAG" == "Linux-x86_64-manylinux" || "$ENV_TAG" == "macOS-arm64-" ]]; then
+            echo "CIBW_BUILD=cp*-$MANYLINUX*" >> $GITHUB_OUTPUT
+          else
+            echo "CIBW_BUILD=cp310-$MANYLINUX* cp313-$MANYLINUX*" >> $GITHUB_OUTPUT
+          fi
```

Comment:
For some reason this isn't picked up by the actual build step - it tests against all versions regardless. To be investigated.